synapse/event_auth.py

```diff
@@ -168,13 +168,23 @@ async def check_state_independent_auth_rules(
         return
 
     # 2. Reject if event has auth_events that: ...
-    auth_events = await store.get_events(
-        event.auth_event_ids(),
-        redact_behaviour=EventRedactBehaviour.as_is,
-        allow_rejected=True,
-    )
     if batched_auth_events:
-        auth_events.update(batched_auth_events)
+        auth_events = dict(batched_auth_events)
```

Comment:
I keep wondering why we're doing this:

```suggestion
        # Copy the batched auth events to avoid mutating them.
        auth_events = dict(batched_auth_events)
```